### PR TITLE
fix(ncnn): update innerproduct so that it does not pack data

### PR DESCRIPTION
**Pull request by fantes**
this PR superseeds https://github.com/jolibrain/deepdetect/pull/1152
(ie it incorporates this PR1152 + fixes another issue)
this should fix https://github.com/jolibrain/deepdetect/issues/1151
issue fixed :  ncnn now uses a lot of architecture dependent optimizations , and our patch on innerproducts for keeping h dimension is implemented only for generic code thus we force generic code execution ; before this patch, data was packed for some architecture optim but generic code was called. This fix by disabling data packing for innerproduct layer

now with UT ! :)

### CODE REVIEW

**Comment by beniz** [comment]
So this subsumes #1152 right ?

File: tests/ut-ncnnapi.cc [hunk 1]
```diff
@@ -299,3 +300,33 @@ TEST(ncnnapi, service_lstm)
...
+                            + ocr_repo + "word_ocr.jpg\"]}";
+  joutstr = japi.jrender(japi.service_predict(jpredictstr));
+  JDoc jd;
+  std::cout << "joutstr=" << joutstr << std::endl;
+  jd.Parse<rapidjson::kParseNanAndInfFlag>(joutstr.c_str());
+  ASSERT_TRUE(!jd.HasParseError());
+  ASSERT_EQ(200, jd["status"]["code"]);
+  ASSERT_TRUE(jd["body"]["predictions"].IsArray());
+  ASSERT_TRUE(jd["body"]["predictions"].Size() == 1);
+  ASSERT_TRUE(jd["body"]["predictions"][0]["classes"][0]["cat"] == "beleved");
```
Comment (beniz) [inline tests/ut-ncnnapi.cc:331]: beleved ? :)

**Review by beniz — APPROVED** [review]

File: tests/ut-ncnnapi.cc (hunk 1 above)
Comment (fantes) [reply tests/ut-ncnnapi.cc:331]: yup, using previous ncnn (ie our old fork) I also get "beleved", while using caffe I get "believed"

Comment (beniz) [reply tests/ut-ncnnapi.cc:331]: Let's try with fp32 activated (fp16 is default), would be surprised it corrects the output though...

**Review by BynaryCobweb — APPROVED** [review]

**Comment by sileht** [comment]
@Mergifyio rebase

**Comment by fantes** [comment]
@Mergifyio rebase

**Pull request merged**